### PR TITLE
CMR-9795: Fixing Linter issues for umm-spec-lib

### PR DESCRIPTION
**Pull request by eereiter**
# Overview

### What is the feature/fix?

Fixing Linter issues for umm-spec-lib

### What is the Solution?

Fixing Linter issues for umm-spec-lib

### What areas of the application does this impact?

umm-spec-lib

# Checklist

- [ x] I have updated/added unit and integration tests that prove my fix is effective or that my feature works
- [ x] New and existing unit and int tests pass locally and remotely with my changes
- [ x] I have removed unnecessary/dead code and imports in files I have changed
- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/additional_attribute.clj [hunk 1]
```diff
@@ -19,33 +19,33 @@
...
+
+;(defmethod parse-value "BOOLEAN"
+;  [data-type ^String value]
+;  (when value
+;    (case value
+;      "true" true
+;      "false" false
+;      "1" true
+;      "0" false
+;      :else (errors/internal-error! (format "Unexpected boolean value [%s]" value)))))
```
Comment (jaybarra) [inline umm-spec-lib/src/cmr/umm_spec/additional_attribute.clj:48]: The clojure style guide would recommend using `#_` over the literal comment tags to disable values.

Comment (eereiter) [reply umm-spec-lib/src/cmr/umm_spec/additional_attribute.clj:48]: Thanks Jay, I was just making sure that what I changed won't break anything.  All of the functions that are commented out are all going to be deleted here in a few minutes.

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version/core.clj [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-   [clojure.test :refer :all]
+   [clojure.test :refer [deftest is]]
    [cmr.umm-spec.migration.version.core :as vm]
    [cmr.umm-spec.versioning :as v]))
 
 
 (deftest test-version-steps
   (with-bindings {#'v/versions {:collection ["1.14" "1.15" "1.15.1" "1.15.2"
-                                             "1.15.3" "1.15.4" "1.15.5" "1.16"]}}
-    (is (= [] (#'vm/version-steps :variable "1.14" "1.14")))
```
Comment (jmaeng72) [inline umm-spec-lib/test/cmr/umm_spec/test/migration/version/core.clj:11]: why this change?

**Review by TylerHeald1 — APPROVED** [review]

**Review by jmaeng72 — APPROVED** [review]

File: umm-spec-lib/test/cmr/umm_spec/test/migration/version/core.clj (hunk 2 above)
Comment (eereiter) [reply umm-spec-lib/test/cmr/umm_spec/test/migration/version/core.clj:11]: Originally this namespace had a bad namespace name - so this test never ran. 
When I ran it, it didn't work.  The with-bindings changes the vector of the collection from its original definition.  Since the definition is for collections the :variable var didn't return anything, so I had to change this test to get it to run and then pass.
